src/js/options.js

```diff
@@ -0,0 +1,41 @@
+
+function requestPermissions() {
+  const checkbox = document.querySelector("#bookmarksPermissions");
+  if (checkbox.checked) {
+    browser.permissions.request({permissions: ["bookmarks"]}).
+    then((response) => {
+      if (response) {
+        browser.runtime.sendMessage({ method: "resetBookmarksContext" });
+      } else {
+        checkbox.checked = false;
+      }
+    }).
+    catch((err) => {
+      return err.message;
```

Comment:
hah, this is only here because the linter was requiring it. Switching to async/await doesn't have this issue with the linter, and it seems like everyone agrees it is more straightforward, so it seems like that is the way to go.